content/5_Plat/Sqrt.mdx

```diff
@@ -29,116 +29,174 @@ approaches. Both run in $\mathcal{O}((N+Q)\sqrt N)$ time.
 <!-- partitioning into sqrt blocks (each block can have some sort of data structure ...)
  -->
 
-See the first example in CPH.
+We partition the array into $\sqrt N)$ blocks, where each block covers an
+interval of size $\lfloor \sqrt{N} \rfloor$. For each block, there exists a
+corresponding `update` and `query` operation.
+
+**Update Queries:** $\mathcal{O}(1)$
+
+To update an element at location $x$, we can find the corresponding block using
+the formula $\frac{x}{\texttt{blocksz}}$, where
+$\texttt{blocksz} = \lfloor \sqrt{N} \rfloor$ and apply the corresponding
+difference between the element currently stored at $x$, and the element we want
+to change it to.
+
+**Sum Queries:** $\mathcal{O}(\sqrt{N})$
+
+To perform a sum query from $[0\ldots r]$, calculate
+
+$$
+\sum_{i = 0}^{R-1} \texttt{blocks}[i] + \sum_{R \cdot \texttt{blocksz}}^r \texttt{nums}[i]
+$$
+
+where $\texttt{blocks}[i]$ represents the total sum of the $i$-th block and the
+$i$-th block represents the sum of the elements from the range
+$[i\sqrt{N},{(i + 1)}\sqrt{N})$, and $R$ represents
+$\frac{r}{\texttt{blocksz}}$.
+
+Finally, $\sum_{i=l}^{r} \texttt{nums}[i]$ is the difference between the two
+prefix sums $\sum_{i=0}^{r}\texttt{nums}[i]$ and
+$\sum_{i=0}^{l-1}\texttt{nums}[i]$ queries for the quantity
```

Comment:
isn't everything from "for the quantity" onwards redundant?

Reply:
yeah